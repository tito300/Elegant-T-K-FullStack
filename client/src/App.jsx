import React, { Component } from 'react';
import { Route, Switch, Redirect } from 'react-router-dom'
import MainPage from './main_page/mainPage';
import Header from './common/pageHeader';
import Footer from './common/footer';
import CartContent from './cart_page/2-cartContent';
import Products from './products_page/productsDisplay';
import LoginIcon from './common/loginIcon';
import Login from './login_page/login';
import './styles/scss/main.scss'
import Product from './products_page/productDisplay';


class Website extends Component {
    state = {

    }

    incrementCount = itemId => {
        const newItems = this.state.cartItems.map(item => {
            if (item.id === itemId) {
                item.count++; // eslint-disable-line
                return item;
            }
            return item;
        });
        this.setState({ cartItems: [...newItems] });
    };


    render() {
        return (
            <React.Fragment>
                <Header />
                <LoginIcon />
                <Switch>
                    <Product path="/product/:id" component={Product} />
                    <Route path="/products" component={Products} />
                    <Route path="/cart/main" render={(props) =>
                        <CartContent
                            {...props}
                            incrementCount={this.incrementCount}
                        />
                    } />
                    <Route path="/login" component={Login} />
                    <Route path="/" component={MainPage} />
                </Switch>
                <Footer />

            </React.Fragment>
        );
    }
}

export default Website;