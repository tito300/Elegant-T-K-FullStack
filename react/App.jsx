import React, { Component } from 'react';
import Header from './common/pageHeader';
import '../public/styles/scss/main.scss'


class App extends Component { // eslint-disable-line
  render() {
    return (
      <Header />
    );
  }
}

export default App;
