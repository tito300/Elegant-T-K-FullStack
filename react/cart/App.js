import React, { Component } from 'react';
import CartPage from './components/1-cartPage';
// import '../../public/styles/scss/main.scss'

class App extends Component { // eslint-disable-line
  render() {
    return (
      <div className="App">
        <CartPage />
      </div>
    );
  }
}

export default App;
