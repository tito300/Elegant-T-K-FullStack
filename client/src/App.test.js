import React from 'react';
import { shallow } from 'enzyme'
import App from './App';
// import Header fro './common/pageHeader';


it('renders without crashing', () => {
   const wrapped = shallow(<App/>);

   expect(wrapped).toMatchSnapshot();
});
