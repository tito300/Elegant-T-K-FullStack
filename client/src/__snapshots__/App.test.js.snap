// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<Fragment>
  <Connect(pageHeader) />
  <Connect(withRouter(LoginIcon)) />
  <Switch>
    <ProductContainer
      component={[Function]}
      path="/product/:id"
    />
    <Route
      path="/products"
      render={[Function]}
    />
    <Route
      path="/cart/main"
      render={[Function]}
    />
    <Route
      component={[Function]}
      path="/login"
    />
    <Route
      component={[Function]}
      path="/"
    />
  </Switch>
  <Footer />
</Fragment>
`;
